lib/include/jxl/decode.h

```diff
@@ -238,6 +243,42 @@ typedef enum {
    * image and always before JXL_DEC_FULL_IMAGE.
    */
   JXL_DEC_JPEG_RECONSTRUCTION = 0x2000,
+
+  /** Informative event by JxlDecoderProcessInput: The header of a box of the
```

Comment:
The data we get in the BoxBuffer would be meant to be decompressed by the application or not based on   the value that can be set with `JxlDecoderSetBoxModeRaw`; so its default value is very important. This is like having a default value for the JxlPixelFormat of RGB 8bit. I think this API would be more robust (harder to use wrong) if we don't have a reliance on this default value behavior, for example if the "raw mode" is set at the time you set the buffer like in the the pixel format case. WDYT?

Reply:
I think if you subscribe to the event, in most applications you'll want to get the uncompressed blob. The exception would be bitstream muxing tools that want to copy boxes as-is.

Reply:
I considered that, the difficulty is that you may have to call `JxlDecoderSetBoxBuffer` multiple times, because you don't know the size beforehand. So that means you need to pass this parameter multiple times and keep it in sync. It's better if you make the decision once, when you just get the JXL_DEC_BOX event. In fact, having to set it at each `JxlDecoderSetBoxBuffer` requires the user to keep extra state for this, which is too much.

A solution is to require calling `JxlDecoderSetBoxModeRaw` after each JXL_DEC_BOX event, right before the first `JxlDecoderSetBoxBuffer` that belongs to this box

WDYT?

Reply:
If you make the default raw, basically forcing the user to call `JxlDecoderSetBoxModeRaw()`, I think it would be ok to remember the last state. I would perhaps use a different name than "box mode raw", something like JxlDecoderDecompressBoxes(true/false).

Reply:
Done, and remembers the setting until decoder reset now